### PR TITLE
Add assertArray function to log.js

### PR DESCRIPTION
**Pull request by cathyxz**
1. Add `assertArray`
2. Add tests
3. Update usage of asserts followed by type casting

### CODE REVIEW

**Comment by erwinmombay** [comment]
you're going to need to update these too:
https://github.com/ampproject/amphtml/blob/master/build-system/babel-plugins/babel-plugin-transform-amp-asserts/index.js#L26-L42

**Comment by erwinmombay** [comment]
and the java code too: https://github.com/ampproject/amphtml/blob/master/build-system/runner/src/org/ampproject/AmpCommandLineRunner.java#L52-L56

**Comment by cathyxz** [comment]
Done. Thanks @erwinmombay for review! I was wondering does this make sense to do? It's an abbreviation / convenience that makes code a bit shorter and easier to read. It's convenient because this assumption allows us to directly do array operations, but I do realise that in a lot of cases, we generally also want to know the `foo` in `!Array<foo>`, and there isn't exactly a good / easy way to do that.

**Comment by erwinmombay** [comment]
yeah i think its a nice convenience for sure, especially since we have other methods of this kind

File: build-system/runner/src/org/ampproject/AmpCommandLineRunner.java [hunk 1]
```diff
@@ -52,7 +52,7 @@
   ImmutableMap<String, Set<String>> suffixTypes = ImmutableMap.of(
       "module$src$log.dev", ImmutableSet.of(
           "assert", "fine", "assertElement", "assertString",
-          "assertNumber", "assertBoolean"),
+          "assertNumber", "assertBoolean", "assertArray"),
```
Comment (erwinmombay) [inline build-system/runner/src/org/ampproject/AmpCommandLineRunner.java]: you're going to need to recompile the runner.jar file to get this to work. LMK if you need help (or i can just compile it for you too)

Comment (cathyxz) [reply build-system/runner/src/org/ampproject/AmpCommandLineRunner.java]: Do we have instructions on how to do that anywhere?

**Comment by cathyxz** [comment]
Looks like we're going to run into a merge conflict with https://github.com/ampproject/amphtml/pull/20056 as well.

**Pull request merged**